C/common/datapoint.cpp

```diff
@@ -122,17 +135,40 @@ DatapointValue::~DatapointValue()
 {
 	if (m_type == T_STRING)
 	{
-		delete m_value.str;
-		m_value.str = NULL;
+		if (m_value.str) {
+			delete m_value.str;
+			m_value.str = NULL;
+		}
 	}
 	if (m_type == T_FLOAT_ARRAY)
 	{
-		delete m_value.a;
-		m_value.a = NULL;
+		if (m_value.a) {
+			delete m_value.a;
+			m_value.a = NULL;
+		}
 	}
-	// For nested DPV, d'tor is always called from holding Datapoint object's destructor
-}
+	// Check for T_DP_DICT or T_DP_LIST
```

Comment:
Is this destructor and DatapointValue::deleteNestedDPV() exactly same now?

Reply:
Yes, they look similar right now